### DISCUSSION TITLE
Docker Directus 9: How to change env variable inside the container?

### DISCUSSION THREAD

**Discussion by khunsatant** [directus/directus#3763] [Archive]
Hello,
I use directus9 rc.28 docker version. If I want to change some config e.g. database config etc. and restart directus, 
how can I do it?
Where is the env file and command to restart?

PS. for the normal version, I just edit .env file then close cmd dialog and run "npx directus start".
Thankyou,
Khunsatant

**Comment by philleepflorence** [comment]
@khunsatant - Did you install Directus via the documentation on [Installing with Docker](https://docs.directus.io/guides/installation/docker.html#docker-compose)?

**Reply by khunsatant** [reply]
yes, I use docker-compose and copy code from that document.

**Reply by staminna** [reply]
@philleepflorence I am also looking for same question to setup emails inside docker, but using a .env file. I don't want to change any setting inside docker image.

**Reply by br41nslug** [reply]
@staminna Environment variables can be managed with a docker-compose file https://docs.directus.io/self-hosted/installation/docker.html#docker-compose

**Comment by s-en-o** [comment]
Hey @khunsatant did you find the answer to your question? I have been dealing with this in the last few days and haven't been able to solve it.

**Reply by paescuj** [reply]
See the example docker compose file under https://docs.directus.io/self-hosted/docker-guide.html#example-docker-compose where the environment variables are defined under `environment:`.

**Reply by s-en-o** [reply]
Thanks @paescuj, I did try set the environment under secrets > environment but no luck. Do I need to set the `.env` file in the `docker-compose.yml` also?

**Reply by pspierce** [reply]
You have to rebuild after you change an environment value.  

e.g., docker compose build directus && docker compose up directus -d

**Reply by paescuj** [reply]
It's enough to just run `docker compose up -d` after changing properties under `environment`. This will be detected by `docker compose` and container(s) getting restarted.

**Comment by pspierce** [comment]
You would think; however, that has not been my experience.

Scott Pierce

**Reply by paescuj** [reply]
From https://docs.docker.com/engine/reference/commandline/compose_up/:
> If there are existing containers for a service, and the service’s configuration or image was changed after the container’s creation, docker compose up picks up the changes by stopping and recreating the containers (preserving mounted volumes).
